client/reader/full-post/index.jsx

```diff
@@ -230,6 +229,34 @@ FullPostView = React.createClass( {
 				</article>
 			</div>
 		);
+	},
+
+	renderFullPostContent: function( content, siteId ) {
+		/**
+		 * Reconstruct the shortcode for the GalleryView by parsing the content string.
+		 */
+		const parsedHTML = $.parseHTML( content );
```

Comment:
As you noted in your comment, `DOMParser` could probably be a suitable alternative here ([caniuse](http://caniuse.com/#feat=xml-serializer)).